homeassistant/components/fritzbox/climate.py

```diff
@@ -135,14 +135,16 @@ def target_temperature(self) -> float:
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        if kwargs.get(ATTR_HVAC_MODE) is not None:
-            hvac_mode = kwargs[ATTR_HVAC_MODE]
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE)
+        if hvac_mode == HVACMode.OFF:
             await self.async_set_hvac_mode(hvac_mode)
-        elif kwargs.get(ATTR_TEMPERATURE) is not None:
-            temperature = kwargs[ATTR_TEMPERATURE]
+        elif target_temp is not None:
             await self.hass.async_add_executor_job(
-                self.data.set_target_temperature, temperature
+                self.data.set_target_temperature, target_temp
             )
+        else:
+            return
         await self.coordinator.async_refresh()
```

Comment:
The else feels weird

Reply:
this should avoid an useless coordinator refresh, in case nothing has changed - this can happen, when the action is called with only `target_temp_high` and `target_temp_low` which is not supported by the device, but still valid for the [`climate.set_temperature`](https://www.home-assistant.io/integrations/climate/#action-climateset_temperature) action

Reply:
maybe raising an error (_instead of just return_) is an alternative here?

Reply:
I guess we can raise a ServiceValidationError yes

Reply:
or something like this at the beginning of the `async_set_temperature`?
```python
        if target_temp is None:
            return
```

Reply:
That can also work, I think I have seen both around

Reply:
but https://github.com/home-assistant/core/pull/126072#discussion_r1763351343 will do exactly the same, as the current code 😀

Reply:
Raising ServiceValidationError is correct.

Reply:
ok, will add it later this evening/night 👍

Reply:
I think this is solved with #126692?